docfx_yaml/extension.py

```diff
@@ -756,6 +764,15 @@ def convert_module_to_package_if_needed(obj):
                     raise ValueError("Unable to dump object\n{0}".format(yaml_data)) from e
 
             file_name_set.add(filename)
+           
+            # Parse the name of the object.
+            # "Types" class will need additional parsing to de-duplicate their names and contain
+            # a portion of their parent name for better disambiguation.
+            node_name = obj['name']
+            if node_name == "types":
+                node_name = ".".join(obj.get('uid').split(".")[-2:])
```

Comment:
Are we sure the last two elements always contain what we want?

Reply:
The name is represented in the last portion of the uid, and I just need one more tiebreaker, which the second last item will work out: we can have same module name but different package (`storage_v1.client`, `storage_v1beta2.client`) or different module and same package (`storage_v1.client`, `storage_v1.types`). Any more different than levels of two should appear under different directory structure, and that doesn't need a disambiguation.

However I should fix hardcoding types, and move this algorithm elsewhere so it detects duplicate entries beside `types`.